imap_processing/mag/l1a/mag_l1a_data.py

```diff
@@ -429,6 +479,478 @@ def to_signed16(n: int) -> int:
                 secondary_vectors.append(vector)
 
         return (
-            np.array(primary_vectors, dtype=np.int64),
-            np.array(secondary_vectors, dtype=np.int64),
+            np.array(primary_vectors, dtype=np.int32),
+            np.array(secondary_vectors, dtype=np.int32),
+        )
+
+    @staticmethod
+    def process_compressed_vectors(  # noqa: PLR0912, PLR0915
+        vector_data: np.ndarray, primary_count: int, secondary_count: int
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Given raw compressed packet data, process into Vectors.
+
+        To do this, we need to decode the compressed data. The compressed data starts
+        with an 8 bit header that defines the width of the uncompressed vectors and
+        if there is a range data section. Then, the vector data follows, then the range
+        data section if it exists.
+
+        To decode, we start by decoding the first compression_width bits. This is an
```

Comment:
Yeah, sorry, this sentence is kind of weird. The compression width is the first 6 bits of the header, which corresponds to a bit count that indicates the width of uncompressed vectors. 

So, if that compression width was `01 0000`, then we have uncompressed vectors values of a width of 16 bits. The full width of the first vector is 16 * 3 + 2 bits, because there are three values per vector and 2 bits for the range value.